### ISSUE TITLE
Biomes o' Plenty trim materials produce no/missing texture

### ISSUE THREAD

**Issue by DJBaphomet** [Andrew6rant/Dynamic-Trim#12]
Biomes o' Plenty allows you to apply 'Glowworm Silk' and 'Rose Quartz Chunk' as materials for a trim's palette. 
Normally, they have no trimmed armour texture in BoP + Vanilla, which I was hoping this mod could fix, however it doesn't seem to do so.
![image](https://github.com/Bawnorton/Dynamic-Trim/assets/44253613/539cce33-753f-42a8-be15-1c0fc463ddc3)

BoP _does_ have a resource pack available to fix this though with vanilla, which works well on its own.
![image](https://github.com/Bawnorton/Dynamic-Trim/assets/44253613/6242a970-6c56-4ced-acc7-0348902c52f4)

However, with the mod enabled, this causes all the BoP trim materials to turn the armour pieces into missing textures. The normal vanilla materials function fine, but it makes the problem with BoP's trim materials even worse than they were originally
![image](https://github.com/Bawnorton/Dynamic-Trim/assets/44253613/dabacfcb-efb8-4107-ba0b-c7afa752ba0c)

This means that you can't use this mod with Biomes o' Plenty's trim materials with the resource pack that's meant to fix BoP's problem, and either have to skip out on using this mod, or deal with armour not appearing as trimmed in your inventory if you wanna use BoP's materials

Hoping for this mod to support Biomes o' Plenty's trim materials so that the resourcepack isn't needed!

Edit to the main post!
Forstride looked at the log in my reply below and mentioned that replacing the first section of the blocks.json file in the Resourcepack with the [first part](https://github.com/Bawnorton/Dynamic-Trim/blob/arch-port/common/src/main/resources/assets/minecraft/atlases/blocks.json) of Dynamic Trim's blocks.json (the part above "palette_key" line) could fix the problem with the resourcepack

And, after editing the blocks.json, it works just fine!
![image](https://github.com/Bawnorton/Dynamic-Trim/assets/44253613/f6aae73a-6cba-470e-8878-75e2535154af)

I'm gonna have this issue stand for now, so people can find a solution to the incompat, and also stay as a suggestion for built-in compat for Dynamic Trims to support BoP's trim materials by itself

**Comment by DJBaphomet** [comment]
Should note: Reported and linked this in Biomes o' Plenty's discord, so they know as well
Fortstride reminded me I could probably see what was going on in the log, so here's that too
https://mclo.gs/5k2xsDJ

**Comment by Bawnorton** [comment]
I will look into dynamically modifying loaded block atlases for trim permutations to move over to a group permutation to allow for Dynamic Trims to be compatible with mods that add other materials.

**Comment by Bawnorton** [comment]
<img width="207" alt="Screenshot 2024-01-10 at 3 02 48 PM" src="https://github.com/Bawnorton/Dynamic-Trim/assets/18416784/b5ca5a1b-bdc6-4304-9e2a-5a368032c590">
<img width="433" alt="Screenshot 2024-01-10 at 3 02 58 PM" src="https://github.com/Bawnorton/Dynamic-Trim/assets/18416784/c8731e8a-b73e-47ff-b4b1-1d1f18258405">

No edits were made to the resource pack and it works now. The resource pack is still needed because of how Mojang implemented trims but at least it works as intended w/ this mod now.

<details>
<summary>The Implementation</summary>

```java
BlocksAtlas atlas = JsonHelper.fromJson(original, BlocksAtlas.class);
atlas.getPalettedPermutationsSource("trims/color_palettes/trim_palette").ifPresent(source -> atlas.addSource(
        source.copy()
              .withType("group_permutations")
              .withDirectories(List.of(
                      "trims/items/helmet",
                      "trims/items/chestplate",
                      "trims/items/leggings",
                      "trims/items/boots"
              ))
              .withTextures(null)
));
```
</details>

Fixed in 1.4.0
